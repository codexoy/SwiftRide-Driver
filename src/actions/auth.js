import {
	ROOT_URL,
	TOKEN_KEY,
	REFRESH_TOKEN_KEY,
	DEFAULT_ERROR,
	signUpRoles
} from "../consts"
import { Linking } from "react-native"
import Storage from "../services/Storage"
import { LOGIN, LOGOUT, API_ERROR } from "../reducers/consts"
import { fetchOrError } from "./utils"

const loginOrRegister = async (endpoint, params, dispatch, callback) => {
	const resp = await dispatch(fetchOrError(endpoint, params))
	if (resp) {
		await setTokens(resp.json.auth_token, resp.json.refresh_token)
		await dispatch(setUser(resp.json.user))
		await callback(resp.json.user)
	} else await callback(undefined)
}

export const directLogin = (email, password, callback) => {
	return async dispatch => {
		const requestParams = {
			method: "POST",
			body: JSON.stringify({ email, password })
		}
		await loginOrRegister(
			"/login/direct",
			requestParams,
			dispatch,
			callback
		)
	}
}

export const register = (params, callback, role = "") => {
	return async dispatch => {
		var data = new FormData()
		Object.keys(params).forEach(key => data.append(key, params[key]))
		const requestParams = {
			method: "POST",
			headers: {
				"Content-Type": "multipart/form-data"
			},
			body: data
		}
		const newCallback = async user => {
			if (user) {
				let roleRequest
				if (role == signUpRoles.student) {
					roleRequest = await dispatch(
						fetchOrError(
							"/user/make_student?teacher_id=" +
								params.teacher_id,
							{
								method: "GET"
							}
						)
					)
				} else if (role == signUpRoles.teacher) {
					roleRequest = await dispatch(
						fetchOrError("/user/make_teacher?", {
							method: "POST",
							body: JSON.stringify({
								price: params.price,
								lesson_duration: params.duration
							})
						})
					)
				}
				if (roleRequest) {
					await callback(user)
					return
				}
			}
			await callback(undefined)
		}

		await loginOrRegister(
			"/login/register",
			requestParams,
			dispatch,
			newCallback
		)
	}
}

const setTokens = async (token, refresh_token) => {
	await Storage.setItem(TOKEN_KEY, token, true)
	await Storage.setItem(REFRESH_TOKEN_KEY, refresh_token, true)
}

export const setUser = user => {
	return dispatch => {
		dispatch({ type: LOGIN, user: user })
	}
}

export const logout = (callback = async () => {}) => {
	return async dispatch => {
		console.log("Logging out")
		await Storage.removeItem(TOKEN_KEY, true)
		await Storage.removeItem(REFRESH_TOKEN_KEY, true)
		await callback()
		setTimeout(() => {
			dispatch({ type: LOGOUT })
		}, 100)
	}
}

export const fetchUser = (callback = () => {}) => {
	/* effectively checks if the token is valid */
	return async dispatch => {
		const resp = await dispatch(
			fetchOrError("/user/me", { method: "GET" }, false)
		)
		if (resp) {
			dispatch(setUser(resp.json.user))
			await callback(resp.json.user)
		} else await callback(undefined)
	}
}

export const exchangeToken = (token, callback) => {
	return async dispatch => {
		try {
			const resp = await dispatch(
				fetchOrError("/login/exchange_token", {
					method: "POST",
					body: JSON.stringify({
						exchange_token: token
					})
				})
			)
			await setTokens(resp.json.auth_token, resp.json.refresh_token)
			await dispatch(fetchUser(callback))
		} catch (error) {
			dispatch({ type: API_ERROR, error: DEFAULT_ERROR })
		}
	}
}

export const openFacebook = (token = "") => {
	let url = ROOT_URL + "/login/facebook"
	if (token) {
		url += "?token=" + token
	}
	Linking.openURL(url)
}
