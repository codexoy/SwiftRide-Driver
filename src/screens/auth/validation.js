import { strings } from "../../i18n"

export const loginValidation = {
	email: {
		email: {
			message: "^" + strings("signin.invalid_email")
		}
	},

	password: {
		length: {
			minimum: 5,
			message: "^" + strings("signin.invalid_password")
		}
	}
}

export const registerValidation = {
	email: {
		email: {
			message: "^" + strings("signin.invalid_email")
		}
	},

	name: {
		length: {
			minimum: 1,
			message: "^" + strings("signin.invalid_field")
		}
	},

	area: {
		length: {
			minimum: 1,
			message: "^" + strings("signin.invalid_field")
		}
	},

	password: {
		length: {
			minimum: 5,
			message: "^" + strings("signin.invalid_email")
		}
	}
}
