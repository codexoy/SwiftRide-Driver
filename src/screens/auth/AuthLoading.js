import React from "react"
import { ActivityIndicator, StyleSheet, StatusBar, View } from "react-native"
import Storage from "../../services/Storage"
import { connect } from "react-redux"
import { TOKEN_KEY } from "../../consts"
import { loadFetchService } from "../../actions/utils"
import { fetchUser, logout } from "../../actions/auth"

export class AuthLoading extends React.Component {
	constructor(props) {
		super(props)
		this._bootstrapAsync()
	}

	// Fetch the token from storage then navigate to our appropriate place
	_bootstrapAsync = async () => {
		this.props.dispatch(loadFetchService())
		await this.props.dispatch(
			fetchUser(async (user = null) => {
				if (user === null) await this.props.dispatch(logout())
				// logging out just to make sure
				this.props.navigation.navigate(user ? "App" : "Auth")
			})
		)
	}

	// Render any loading content that you like here
	render() {
		return (
			<View style={styles.container}>
				<ActivityIndicator />
				<StatusBar barStyle="default" />
			</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "center"
	}
})

export default connect()(AuthLoading)
