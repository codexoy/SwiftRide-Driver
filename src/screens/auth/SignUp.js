import React, { Fragment } from "react"
import {
	View,
	Text,
	StyleSheet,
	ScrollView,
	KeyboardAvoidingView,
	TouchableOpacity,
	Platform,
	Keyboard
} from "react-native"
import { connect } from "react-redux"
import { register } from "../../actions/auth"
import { API_ERROR, POP_ERROR } from "../../reducers/consts"
import validate, { registerValidation } from "../../actions/validate"
import { strings } from "../../i18n"
import AuthInput from "../../components/AuthInput"
import { MAIN_PADDING, DEFAULT_MESSAGE_TIME, DEFAULT_IMAGE } from "../../consts"
import { Icon } from "react-native-elements"
import LoadingButton from "../../components/LoadingButton"
import SlidingMessage from "../../components/SlidingMessage"
import { popLatestError } from "../../actions/utils"
import UploadProfileImage from "../../components/UploadProfileImage"

export class SignUp extends React.Component {
	constructor(props) {
		super(props)
		this.register = this.register.bind(this)
		this.role = this.props.navigation.getParam("role")
		this.state = {
			api_error: "",
			slidingMessageVisible: false,
			imageError: "",
			image: ""
		}
		this.inputs = {
			email: {},
			name: { iconName: "person", placeholder: strings("signup.name") },
			area: {
				iconName: "person-pin",
				placeholder: strings("signup.area")
			},
			password: { secureTextEntry: true, iconName: "security" }
		}
		Object.keys(this.inputs).forEach(input => {
			this.state[input] = ""
			this.state[input + "Error"] = ""
		})
	}
	componentDidMount() {
		if (Platform.OS === "android") {
			this.keyboardEventListeners = [
				Keyboard.addListener(
					"keyboardDidShow",
					this._handleKeyboardShow
				),
				Keyboard.addListener(
					"keyboardDidHide",
					this._handleKeyboardHide
				)
			]
		}
	}

	componentWillUnmount() {
		this.keyboardEventListeners &&
			this.keyboardEventListeners.forEach(eventListener =>
				eventListener.remove()
			)
	}

	_handleKeyboardHide = () => {
		this.scrollView.scrollTo({ y: 0 })
	}

	_handleKeyboardShow = () => {
		this.scrollView.scrollToEnd()
	}

	componentDidUpdate() {
		const error = this.props.dispatch(popLatestError(API_ERROR))
		if (error) {
			this.setState({
				error,
				slidingMessageVisible: true
			})
		}
	}

	async register() {
		let error,
			errors = []
		Object.keys(this.inputs).forEach(input => {
			error = validate(input, this.state[input], registerValidation)
			if (error) errors.push(error)
			this.setState({ [input + "Error"]: error })
		})
		if (this.state.image == "") {
			this.setState({ imageError: strings("signup.image_required") })
			return
		}

		if (errors.length > 0) return
		this.button.showLoading(true)

		await this.props.dispatch(
			register(
				{
					email: this.state.email,
					area: this.state.area,
					password: this.state.password,
					name: this.state.name,
					image: this.state.image
				},
				user => {
					this.button.showLoading(false)
					if (user) {
						this.setState(
							{ api_error: "", slidingMessageVisible: true },
							() => {
								setTimeout(
									() => this.props.navigation.navigate("App"),
									DEFAULT_MESSAGE_TIME
								)
							}
						)
					}
				}
			)
		)
	}

	renderInputs = () => {
		return Object.keys(this.inputs).map((name, index) => {
			const props = this.inputs[name]
			return (
				<AuthInput
					key={`key${name}`}
					name={name}
					placeholder={props.placeholder || strings("signin." + name)}
					onChangeText={input => this.setState({ [name]: input })}
					value={this.state[name]}
					testID={`r${name}Input`}
					iconName={props.iconName || name}
					errorMessage={this.state[`${name}Error`]}
					validation={registerValidation}
					secureTextEntry={props.secureTextEntry || false}
				/>
			)
		})
	}

	render() {
		return (
			<View style={styles.container}>
				<SlidingMessage
					visible={this.state.slidingMessageVisible}
					error={this.state.api_error}
					success={strings("signup.success")}
					close={() =>
						this.setState({ slidingMessageVisible: false })
					}
				/>
				<TouchableOpacity
					onPress={() => {
						this.props.navigation.goBack()
					}}
					style={styles.backButton}
				>
					<Icon name="arrow-forward" type="material" />
				</TouchableOpacity>
				<KeyboardAvoidingView
					behavior={Platform.OS === "ios" ? "position" : null}
				>
					<ScrollView
						keyboardDismissMode={
							Platform.OS === "ios" ? "interactive" : "on-drag"
						}
						keyboardShouldPersistTaps="handled"
						ref={r => (this.scrollView = r)}
					>
						<View style={styles.formContainer}>
							<UploadProfileImage
								style={styles.profilePic}
								image={this.state.image.uri || DEFAULT_IMAGE}
								upload={async source => {
									this.setState({
										image: source,
										imageError: ""
									})
								}}
							/>
							<Text style={styles.error}>
								{this.state.imageError}
							</Text>
							{this.renderInputs()}
							<LoadingButton
								title={strings("signup.signup_button")}
								onPress={this.register}
								ref={c => (this.button = c)}
								style={styles.button}
								indicatorColor="#fff"
							/>
						</View>
					</ScrollView>
				</KeyboardAvoidingView>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	error: {
		marginTop: 12,
		color: "red",
		fontSize: 12
	},
	backButton: {
		alignSelf: "flex-start",
		marginTop: 12,
		marginLeft: MAIN_PADDING
	},
	form: {
		flex: 1,
		marginTop: MAIN_PADDING
	},
	button: {
		marginTop: 20
	},
	profilePic: {
		width: 80,
		height: 80,
		borderRadius: 40,
		alignSelf: "center"
	},
	container: {
		flex: 1
	},
	formContainer: {
		flex: 1,
		paddingLeft: MAIN_PADDING,
		paddingRight: MAIN_PADDING,
		alignItems: "center",
		paddingBottom: 40
	}
})

const mapStateToProps = state => {
	return {
		errors: state.errors
	}
}

export default connect(mapStateToProps)(SignUp)
