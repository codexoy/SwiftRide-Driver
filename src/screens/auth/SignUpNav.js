import { createStackNavigator } from "react-navigation"
import SignUpAs from "./SignUpAs"
import { SignUp } from "./SignUp"

export default createStackNavigator(
	{ First: SignUpAs, Second: SignUp },
	{
		initialRouteKey: "First",
		headerMode: "none",
		navigationOptions: {
			headerVisible: false
		}
	}
)
