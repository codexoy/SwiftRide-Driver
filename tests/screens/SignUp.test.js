import { Button } from "react-native"
import React from "react"

import renderer from "react-test-renderer"

import { SignUp } from "../../src/screens/auth/SignUp"

let render
describe("SignUp", () => {
	beforeAll(() => {
		render = renderer.create(<SignUp />)
	})
	test("view renders correctly", () => {
		const tree = render.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
