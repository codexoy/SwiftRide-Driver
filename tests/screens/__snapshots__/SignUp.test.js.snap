// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp view renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "paddingLeft": 26,
          "paddingRight": 26,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 120,
              "width": 120,
            },
          ]
        }
      >
        <FastImageView
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../assets/images/sign_up_success.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginTop": 24,
          }
        }
      >
        נרשמת בהצלחה!
      </Text>
      <Text
        style={
          Object {
            "color": "gray",
            "marginTop": 12,
            "textAlign": "center",
          }
        }
      >
        ההרשמה עברה בהצלחה. כעת אנו ממתינים שהמורה אליו נרשמת יאשר אותך בתור תלמיד.
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(12,116,244)",
            "borderRadius": 28,
            "elevation": 8,
            "height": 56,
            "justifyContent": "center",
            "marginTop": 40,
            "opacity": 1,
            "shadowColor": "rgb(12,116,244)",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.5,
            "shadowRadius": 16,
            "width": 160,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          עבור לעמוד הראשי
        </Text>
      </View>
    </View>
  </Modal>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignSelf": "flex-start",
        "marginLeft": 26,
        "marginTop": 12,
        "opacity": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        underlayColor="white"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <RCTScrollView
        keyboardDismissMode="interactive"
        keyboardShouldPersistTaps="handled"
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "paddingBottom": 40,
                "paddingLeft": 26,
                "paddingRight": 26,
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    Object {
                      "alignSelf": "center",
                      "borderRadius": 40,
                      "height": 80,
                      "width": 80,
                    },
                  ]
                }
              >
                <FastImageView
                  resizeMode="cover"
                  source={
                    Object {
                      "uri": "https://res.cloudinary.com/hsmkon3br/image/upload/c_thumb,g_face,h_80,w_80/v1554188012/678099-profile-filled-256.png",
                    }
                  }
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
            </View>
            <Text
              style={
                Object {
                  "color": "red",
                  "fontSize": 12,
                  "marginTop": 12,
                }
              }
            >
              
            </Text>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderRadius": 28,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginTop": 12,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginLeft": 6,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        theme={
                          Object {
                            "colors": Object {
                              "disabled": "hsl(208, 8%, 90%)",
                              "divider": "#bcbbc1",
                              "error": "#ff190c",
                              "grey0": "#393e42",
                              "grey1": "#43484d",
                              "grey2": "#5e6977",
                              "grey3": "#86939e",
                              "grey4": "#bdc6cf",
                              "grey5": "#e1e8ee",
                              "greyOutline": "#bbb",
                              "platform": Object {
                                "android": Object {
                                  "error": "#f44336",
                                  "primary": "#2196f3",
                                  "secondary": "#9C27B0",
                                  "success": "#4caf50",
                                  "warning": "#ffeb3b",
                                },
                                "ios": Object {
                                  "error": "#ff3b30",
                                  "primary": "#007aff",
                                  "secondary": "#5856d6",
                                  "success": "#4cd964",
                                  "warning": "#ffcc00",
                                },
                              },
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#8F0CE8",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          }
                        }
                        underlayColor="white"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#c9c9c9",
                                "fontSize": 20,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="אימייל"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                      "padding": 16,
                      "textAlign": "right",
                    }
                  }
                  testID="remailInput"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderRadius": 28,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginTop": 12,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginLeft": 6,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        theme={
                          Object {
                            "colors": Object {
                              "disabled": "hsl(208, 8%, 90%)",
                              "divider": "#bcbbc1",
                              "error": "#ff190c",
                              "grey0": "#393e42",
                              "grey1": "#43484d",
                              "grey2": "#5e6977",
                              "grey3": "#86939e",
                              "grey4": "#bdc6cf",
                              "grey5": "#e1e8ee",
                              "greyOutline": "#bbb",
                              "platform": Object {
                                "android": Object {
                                  "error": "#f44336",
                                  "primary": "#2196f3",
                                  "secondary": "#9C27B0",
                                  "success": "#4caf50",
                                  "warning": "#ffeb3b",
                                },
                                "ios": Object {
                                  "error": "#ff3b30",
                                  "primary": "#007aff",
                                  "secondary": "#5856d6",
                                  "success": "#4cd964",
                                  "warning": "#ffcc00",
                                },
                              },
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#8F0CE8",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          }
                        }
                        underlayColor="white"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#c9c9c9",
                                "fontSize": 20,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="שם מלא"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                      "padding": 16,
                      "textAlign": "right",
                    }
                  }
                  testID="rnameInput"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderRadius": 28,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginTop": 12,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginLeft": 6,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        theme={
                          Object {
                            "colors": Object {
                              "disabled": "hsl(208, 8%, 90%)",
                              "divider": "#bcbbc1",
                              "error": "#ff190c",
                              "grey0": "#393e42",
                              "grey1": "#43484d",
                              "grey2": "#5e6977",
                              "grey3": "#86939e",
                              "grey4": "#bdc6cf",
                              "grey5": "#e1e8ee",
                              "greyOutline": "#bbb",
                              "platform": Object {
                                "android": Object {
                                  "error": "#f44336",
                                  "primary": "#2196f3",
                                  "secondary": "#9C27B0",
                                  "success": "#4caf50",
                                  "warning": "#ffeb3b",
                                },
                                "ios": Object {
                                  "error": "#ff3b30",
                                  "primary": "#007aff",
                                  "secondary": "#5856d6",
                                  "success": "#4cd964",
                                  "warning": "#ffcc00",
                                },
                              },
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#8F0CE8",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          }
                        }
                        underlayColor="white"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#c9c9c9",
                                "fontSize": 20,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="איזור למידה"
                  secureTextEntry={false}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                      "padding": 16,
                      "textAlign": "right",
                    }
                  }
                  testID="rareaInput"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "paddingHorizontal": 10,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "borderBottomWidth": 1,
                    "borderColor": "#e0e0e0",
                    "borderRadius": 28,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginTop": 12,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 40,
                      "justifyContent": "center",
                      "marginLeft": 15,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginLeft": 6,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "justifyContent": "center",
                          }
                        }
                        theme={
                          Object {
                            "colors": Object {
                              "disabled": "hsl(208, 8%, 90%)",
                              "divider": "#bcbbc1",
                              "error": "#ff190c",
                              "grey0": "#393e42",
                              "grey1": "#43484d",
                              "grey2": "#5e6977",
                              "grey3": "#86939e",
                              "grey4": "#bdc6cf",
                              "grey5": "#e1e8ee",
                              "greyOutline": "#bbb",
                              "platform": Object {
                                "android": Object {
                                  "error": "#f44336",
                                  "primary": "#2196f3",
                                  "secondary": "#9C27B0",
                                  "success": "#4caf50",
                                  "warning": "#ffeb3b",
                                },
                                "ios": Object {
                                  "error": "#ff3b30",
                                  "primary": "#007aff",
                                  "secondary": "#5856d6",
                                  "success": "#4cd964",
                                  "warning": "#ffcc00",
                                },
                              },
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#8F0CE8",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          }
                        }
                        underlayColor="white"
                      >
                        <Text
                          allowFontScaling={false}
                          style={
                            Array [
                              Object {
                                "color": "#c9c9c9",
                                "fontSize": 20,
                              },
                              Object {
                                "backgroundColor": "transparent",
                              },
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                          testID="iconIcon"
                        >
                          
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <TextInput
                  allowFontScaling={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="סיסמה"
                  secureTextEntry={true}
                  style={
                    Object {
                      "alignSelf": "center",
                      "color": "black",
                      "flex": 1,
                      "fontSize": 18,
                      "minHeight": 40,
                      "padding": 16,
                      "textAlign": "right",
                    }
                  }
                  testID="rpasswordInput"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "opacity": 1,
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgb(12,116,244)",
                    "borderRadius": 32,
                    "height": 64,
                    "justifyContent": "center",
                    "marginTop": 20,
                    "width": "100%",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 20,
                    }
                  }
                >
                  הרשמה
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
